pkg/controller/build/build_controller.go

```diff
@@ -995,15 +995,25 @@ func (ctrl *Controller) prepareForBuild(inputs *buildInputs) (ImageBuildRequest,
 // Determines if we should run a build, then starts a build pod to perform the
 // build, and updates the MachineConfigPool with an object reference for the
 // build pod.
-func (ctrl *Controller) startBuildForMachineConfigPool(ps *poolState) error {
-	inputs, err := ctrl.getBuildInputs(ps)
-	if err != nil {
-		return fmt.Errorf("could not fetch build inputs: %w", err)
-	}
+func (ctrl *Controller) startBuildForMachineConfigPool(mosb *mcfgv1alpha1.MachineOSBuild) error {
+
+	// we need to add osImageURL to mosbuild, will reduce api calls to configmaps
+	// ocb config will live in th mosb
+	// pool will live in the mosb
+	// mc we can get based off the pool specified in the mosb.... though, given how we could use this in two places
+
+	machineOSConfigs, err := ctrl.machineOSConfigLister.List(labels.Everything())
 
-	ibr, err := ctrl.prepareForBuild(inputs)
+	var ourConfig *mcfgv1alpha1.MachineOSConfig
+	for _, c := range machineOSConfigs {
+		if c.Spec.MachineConfigPool.Name == mosb.Spec.MachineConfigPool.Name {
```

Comment:
thought: Since it looks like we're repeating this pattern here and in NodeController, lets add a helper in the `apihelpers` package that does something like:

```golang
func GetMachineOSConfigForPool(moscLister MoscListerInterface, pool *mcfgv1.MachineConfigPool) (*mcfgv1alpha1.MachineOSConfig, error) {
	machineOSConfigs, err := ctrl.machineOSConfigLister.List(labels.Everything())
	if err != nil {
		return nil, err
	}

	for _, c := range machineOSConfigs {
		if c.Spec.MachineConfigPool.Name == mosb.Spec.MachineConfigPool.Name {
			return c, nil
		}
	}

	// https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L144
	return nil, apierrors.NewNotFound(...)
}
```

Because we return an error if we can't find the object, the caller can check for that error and determine what to do next. We should also add an equivalent helper for MachineOSBuilds as well. Although for that particular case, we may want to return a list instead.